odoo/fields.py

```diff
@@ -1890,20 +1891,47 @@ def convert_to_cache(self, value, record, validate=True):
             # the cache must contain bytes or memoryview, but sometimes a string
             # is given when assigning a binary field (test `TestFileSeparator`)
             return value.encode()
+        # TODO SEB this can probably be cleaned because it is useless: the SQL
+        # always returns a string, the only case where we enter this condition
+        # is when the value is False, and then human_size returns False
+        # and convert to cache should be None for consistency when it is False!
         if isinstance(value, int) and \
                 (record._context.get('bin_size') or
                  record._context.get('bin_size_' + self.name)):
             # If the client requests only the size of the field, we return that
             # instead of the content. Presumably a separate request will be done
             # to read the actual content, if necessary.
-            return human_size(value)
+            value = human_size(value)
         return None if value is False else value
 
     def convert_to_record(self, value, record):
         if isinstance(value, _BINARY):
             return bytes(value)
         return False if value is None else value
 
+    def compute_value(self, records):
+        bin_size_name = 'bin_size_' + self.name
+        if records.env.context.get('bin_size') or records.env.context.get(bin_size_name):
+            # always compute without bin_size
+            records_no_bin_size = records.with_context(**{'bin_size': False, bin_size_name: False})
+            super().compute_value(records_no_bin_size)
+            # manually update the bin_size cache
+            cache = records.env.cache
+            for record_no_bin_size, record in zip(records_no_bin_size, records):
+                value = cache.get(record_no_bin_size, self)
```

Comment:
This line can raise a `CacheMiss` if the value has not been assigned by the compute method.
In that case, don't set the cache.

Reply:
Isn't it a programming mistake that the compute method does not assign anything? Do we really want to catch it? I was under the impression that was part of the contract of the new ORM.

I wanted to see if runbot would pass without the `except` (which it does), but I'll reintroduce it if you feel it is safer this way.

Reply:
Ok thank you, I understand now. So in practice it makes no difference, but I like the idea of delegating the processing of the error to the usual methods.